frontend/components/bc/NetworkSelector.vue

```diff
@@ -0,0 +1,56 @@
+<script setup lang="ts">
+import type { MultiBarItem } from '~/types/multiBar'
+import { IconNetwork } from '#components'
+import { ChainInfo, ChainIDs } from '~/types/network'
+
+const props = defineProps<{
+  displayOnly?: boolean
+}>()
+
+const { availableNetworks, isNetworkDisabled } = useNetworkStore()
+
+/** If the v-model is:
+ *  - A ChainIDs: only one network can be selected by the user. Prop `:display-only` must be false or omitted.
+ *  - An array of ChainIDs:
+ *    - and prop `:display-only` is `false`/omitted: several networks can be selected by the user,
+ *    - and prop `:display-only` is `true`: the networks in the array are shown to the user but they are unclickable. */
+const liveState = defineModel<ChainIDs|ChainIDs[]>({ required: false })
```

Comment:
> Would prefer to have the model to the outside be called 'selection' like for the other Bars's.

Done.

> The selection should only be used in editable mode. I would suggest to have an optional prop 'availableNetworks?: ChainIDs[]' ... if this is set then it is used for the button list, else the one from the useNetworkStore

This complicates the code a a lot as you can see.
Also, it was making sense that the list of networks in the readonly mode was coming from the v-model `selection` because we are displaying the selection of the user.
But I changed it as you asked. I suppose that the way to use the component is now more standard. I trust my teacher.